### PR TITLE
JAMES-2578 Fix warning using Mailet new API

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java [hunk 1]
```diff
@@ -694,6 +697,10 @@ public void service(Mail mail) throws MessagingException {
 
     }
 
+    private Attribute makeAttribute(boolean value) {
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java]: these makeAttribute methods are ugly, do we really need them?

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java [hunk 2]
```diff
@@ -674,7 +677,7 @@ public void service(Mail mail) throws MessagingException {
                     if (isDebug()) {
                         LOGGER.debug("OK (by CLAMD on {})", socket.getInetAddress());
                     }
-                    mail.setAttribute(MAIL_ATTRIBUTE_NAME, "false");
+                    mail.setAttribute(makeAttribute(false));
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java]: did you check that potential other mailets using this attribute are expecting a boolean and not a String?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRetriesHelper.java [hunk 3]
```diff
@@ -19,32 +19,31 @@
...
-        mail.setAttribute(DELIVERY_RETRY_COUNT, 0);
+        mail.setAttribute(makeAttribute(0));
     }
 
     public static void incrementRetries(Mail mail) {
-        mail.setAttribute(DELIVERY_RETRY_COUNT, retrieveRetries(mail) + 1);
+        mail.setAttribute(makeAttribute(retrieveRetries(mail) + 1));
     }
 
+    private static Attribute makeAttribute(Integer value) {
```
Comment (rouazana) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRetriesHelper.java]: idem

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java (hunk 1 above)
Comment (blackheaven) [reply mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java]: we do not need them, but I find it more easy to read than specifying to complete `Attribute` definition.

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java (hunk 2 above)
Comment (blackheaven) [reply mailet/standard/src/main/java/org/apache/james/transport/mailets/ClamAVScan.java]: I did, it's only locally used.

**Comment by Arsnael** [comment]
```
[304dce724956602222a15b2f22d003de3fe6fb2d] [ERROR] /james-project/server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRetryHelperTest.java:[69,75] incompatible types: org.apache.mailet.AttributeName cannot be converted to java.lang.String
[304dce724956602222a15b2f22d003de3fe6fb2d] [ERROR] /james-project/server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/remote/delivery/DeliveryRetryHelperTest.java:[77,75] incompatible types: org.apache.mailet.AttributeName cannot be converted to java.lang.String
```

**Comment by blackheaven** [comment]
```
Error response from daemon: No such container: james-server-cassandra-ldap-james-2578-reboot-5-0b918f01-9b57-4bd0-bfd7-50557dd63fa7
Error response from daemon: No such container: james-server-jpa-sample-testingjames-2578-reboot-5-0b918f01-9b57-4bd0-bfd7-50557dd63fa7
```

test this please

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
Finally no more `getAttributes`

**Comment by Arsnael** [comment]
Read it...

test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**